### PR TITLE
add task kanban backend connection

### LINKED ISSUE #499: Create, update and delete tasks on kanban

### PR DESCRIPTION
**Pull request by DasProffi**
## Which issues does this pull request close?
closes #499

### CODE REVIEW

**Comment by DasProffi** [comment]
Blocked by https://github.com/helpwave/services/issues/263

File: tasks/mutations/task_mutations.ts [hunk 1]
```diff
@@ -23,3 +30,151 @@ export type TaskMinimalDTO = {
...
+      }
+
+      const req = new GetTasksByPatientRequest()
+      req.setPatientId(patientID)
+
+      const res = await taskService.getTasksByPatient(req, getAuthenticatedGrpcMetadata())
+
+      if (!res.getTasksList()) {
+        // TODO some check whether request was successful
+        console.error('create room failed')
```
Comment (use-to) [inline tasks/mutations/task_mutations.ts:64]: update error message

File: tasks/mutations/task_mutations.ts [hunk 2]
```diff
@@ -23,3 +30,151 @@ export type TaskMinimalDTO = {
...
+      }
```
Comment (use-to) [inline tasks/mutations/task_mutations.ts:65]:
```suggestion
      }
      
```

File: tasks/mutations/task_mutations.ts [hunk 3]
```diff
@@ -23,3 +30,151 @@ export type TaskMinimalDTO = {
...
+
+      const req = new GetTasksByPatientSortedByStatusRequest()
+      req.setPatientId(patientID)
+
+      const res = await taskService.getTasksByPatientSortedByStatus(req, getAuthenticatedGrpcMetadata())
+
+      if (!res.toObject()) {
+        // TODO some check whether request was successful
+        console.error('create room failed')
+      }
```
Comment (use-to) [inline tasks/mutations/task_mutations.ts:103]: update error message

File: tasks/mutations/task_mutations.ts [hunk 4]
```diff
@@ -23,3 +30,151 @@ export type TaskMinimalDTO = {
...
+        [TaskStatus.TASK_STATUS_IN_PROGRESS]: res.getInProgressList().map(value => mapping(value, TaskStatus.TASK_STATUS_IN_PROGRESS)),
+        [TaskStatus.TASK_STATUS_DONE]: res.getDoneList().map(value => mapping(value, TaskStatus.TASK_STATUS_DONE)),
+      }
+
+      return tasks
+    },
+  })
+}
+
+export const useTaskToToDoMutation = (callback: () => void) => {
```
Comment (use-to) [inline tasks/mutations/task_mutations.ts:131]: assign default noop callback

File: tasks/mutations/task_mutations.ts [hunk 5]
```diff
@@ -23,3 +30,151 @@ export type TaskMinimalDTO = {
...
+      // TODO some check whether request was successful
+
+      queryClient.refetchQueries([tasksQueryKey]).then()
+      callback()
+      return req.toObject()
+    },
+  })
+}
+
+export const useTaskToInProgressMutation = (callback: () => void) => {
```
Comment (use-to) [inline tasks/mutations/task_mutations.ts:148]: assign default noop callback

File: tasks/mutations/task_mutations.ts [hunk 6]
```diff
@@ -23,3 +30,151 @@ export type TaskMinimalDTO = {
...
+      // TODO some check whether request was successful
+
+      queryClient.refetchQueries([tasksQueryKey]).then()
+      callback()
+      return req.toObject()
+    },
+  })
+}
+
+export const useTaskToDoneMutation = (callback: () => void) => {
```
Comment (use-to) [inline tasks/mutations/task_mutations.ts:165]: assign default noop callback

**Review by use-to — CHANGES_REQUESTED** [review]

**Review by use-to — COMMENTED** [review]
Works like a charm. Awaiting task creation 👍🏻

**Review by use-to — APPROVED** [review]
Works like a charm. Awaiting task creation 👍🏻

**Comment by MaxSchaefer** [comment]
@DasProffi @useto the linked issue says "Create, update and delete tasks on kanban" any reason why the creation is not implemented? Or am I doing something wrong? 😅

**Comment by DasProffi** [comment]
Because then the issue would be bloated, since only implementing create makes not much sense when the view has to handle read, write, update.
But also I just over read it

**Pull request merged**